File: ios/Classes/AaaPlugin.swift
import Flutter
import UIKit

class AaaApiImpl:AaaFlutterApi, AaaApi{
    
    init(binaryMessenger: FlutterBinaryMessenger) {
        super.init(binaryMessenger: binaryMessenger)
    }
    
    func bbb(name: String, completion: @escaping (Result<String, Error>) -> Void) {
        print("native bbb(): " + name)
        completion(.success("bbb: " + name))
//        ccc { result in
//            switch result{
//            case .success(()):
//                print("Func ccc Success")
//                return
//            case .failure(let error):
//                print("Func ccc error: " + error.message!)
//            }
//            
//        }
        ddd()

    }
    
    func ddd(){
        ccc{ result in
            switch result{
            case .success(()):
                print("Func ddd->ccc() Success")
            case .failure(let error):
                print("Func ddd->ccc() error: " + error.message!)
            }
        }
    }

}


public class AaaPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let instance = AaaApiImpl(binaryMessenger: registrar.messenger())
        AaaApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance)
    }
}

File: android/src/main/kotlin/com/fff/aaa/AaaPlugin.kt
package com.fff.aaa

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.BinaryMessenger

/** AaaPlugin */
class AaaPlugin : FlutterPlugin, AaaPluginApi.AaaApi {

    private lateinit var binaryMessagener:  io.flutter.plugin.common.BinaryMessenger

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        binaryMessagener = flutterPluginBinding.binaryMessenger
        AaaPluginApi.AaaApi.setUp(binaryMessagener, this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        AaaPluginApi.AaaApi.setUp(binding.binaryMessenger, null)
    }

    override fun bbb(name: String, result: AaaPluginApi.Result<String>) {
        val pluginF: AaaPluginFlutter =  AaaPluginFlutter(binaryMessagener)
        pluginF.callFlutterMetod_ccc()
        result.success("bbb: $name")
    }

}

class Result : AaaPluginApi.VoidResult {
    override fun success() {
    }

    override fun error(error: Throwable) {
    }

}

private class AaaPluginFlutter {
    var pluginFlutter: AaaPluginApi.AaaFlutterApi? = null

    constructor(binding: BinaryMessenger) {
        pluginFlutter = AaaPluginApi.AaaFlutterApi(binding)
    }

    fun callFlutterMetod_ccc() {
        pluginFlutter!!.ccc(Result())
    }
}